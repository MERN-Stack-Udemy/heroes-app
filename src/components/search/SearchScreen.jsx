import React from 'react'

export const SearchScreen = () => {
    return (
        <div>
            <h1>Serar Screen</h1>
            <hr />
            <div className="row">
                <div className="col-5">
                    <h4 > Search Form</h4>
                    <hr />
                    <form action="">
                        <input 
                            type="text" 
                            placeholder="Find your hero"
                            className="form-control"
                        />
                        <button
                            type="submit"
                            className="btn m-1 btn-block btn-primary"
                        >
                            Search
                        </button>
                    </form>
                </div>

            </div>
        </div>
    )
}
