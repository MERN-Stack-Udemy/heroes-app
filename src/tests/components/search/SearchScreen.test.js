import { mount } from "enzyme"
import { MemoryRouter, Route } from "react-router-dom";
import { SearchScreen } from "../../../components/search/SearchScreen";


describe('test on <SearchScreen/> component', () => {
    
    const historyMock = {
        push: jest.fn()
    };

    test('should display correctly with default values', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.alert-info').text().trim()).toBe('Search some Hero');
    });

    test('should display Batman and input have a value from queryString', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        );
        expect( wrapper.find('input').prop('value') ).toBe('batman');
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should display error message if hero not find', () => {
        const heroSearch = 'batmanjsjdls'
        const wrapper = mount(
            <MemoryRouter initialEntries={[`/search?q=${heroSearch}`]}>
                <Route path="/search" component={ SearchScreen } />
            </MemoryRouter>
        );

        expect( wrapper.find('.alert-danger').text()).toBe(`There is no a hero with: '${ heroSearch }'.`);
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('should call history of history', () => {
        const historyMock = {
            push: jest.fn()
        };
        const heroSearch = 'batman';
        const wrapper = mount(
            <MemoryRouter initialEntries={[`/search?q=${heroSearch}`]}>
                <Route 
                    path="/search" 
                    component={() => <SearchScreen history={historyMock} /> } 
                />
            </MemoryRouter>
        );
        // simulate changes into input
        wrapper.find('input').simulate('change', {
            target:{
                name: 'searchText',
                value: heroSearch
            }
        });
        //make form submit
        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        });

        expect( historyMock.push ).toHaveBeenCalledWith(`?q=${ heroSearch }`);
    })
    

})
