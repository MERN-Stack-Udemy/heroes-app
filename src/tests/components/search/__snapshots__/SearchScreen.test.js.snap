// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <SearchScreen/> component should display Batman and input have a value from queryString 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="false"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batman"
        />
        <button
          className="btn m-1 btn-block btn-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results
      </h4>
      <hr />
      <div
        className="card ms-3"
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/heroes/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Batman
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Bruce Wayne
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                   
                  Detective Comics #27
                   
                </small>
              </p>
              <a
                href="/hero/dc-batman"
                onClick={[Function]}
              >
                 Show more. 
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test on <SearchScreen/> component should display correctly with default values 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="false"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn m-1 btn-block btn-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results
      </h4>
      <hr />
      <div
        className="alert alert-info"
      >
        Search some Hero
      </div>
    </div>
  </div>
</div>
`;

exports[`test on <SearchScreen/> component should display error message if hero not find 1`] = `
<div>
  <h1>
    Search Screen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
         Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="false"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value="batmanjsjdls"
        />
        <button
          className="btn m-1 btn-block btn-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
         Results
      </h4>
      <hr />
      <div
        className="alert alert-danger"
      >
        There is no a hero with: '
        batmanjsjdls
        '.
      </div>
    </div>
  </div>
</div>
`;
